### PR TITLE
controller-gen: added tests for crd generation

### PR DESCRIPTION
**Pull request by droot**
Refactored CRD generator to support testing

### CODE REVIEW

File: pkg/crd/generator/testData/.gitignore [hunk 1]
```diff
@@ -0,0 +1,24 @@
+
```
Comment (mengqiy) [inline pkg/crd/generator/testData/.gitignore]: Why we need this file?
Is this file scaffolded?

File: pkg/crd/generator/testData/Gopkg.toml [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+    "k8s.io/code-generator/cmd/deepcopy-gen", # for go generate
+    "sigs.k8s.io/controller-tools/cmd/controller-gen", # for crd/rbac generation
+    "sigs.k8s.io/controller-runtime/pkg/client/config",
+    "sigs.k8s.io/controller-runtime/pkg/controller",
+    "sigs.k8s.io/controller-runtime/pkg/handler",
+    "sigs.k8s.io/controller-runtime/pkg/manager",
+    "sigs.k8s.io/controller-runtime/pkg/runtime/signals",
+    "sigs.k8s.io/controller-runtime/pkg/source",
+    "sigs.k8s.io/testing_frameworks/integration", # for integration testing
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1",
```
Comment (mengqiy) [inline pkg/crd/generator/testData/Gopkg.toml]: nit: please fix indent.

File: pkg/crd/generator/testData/hack/boilerplate.go.txt [hunk 3]
```diff
@@ -0,0 +1,14 @@
+/*
+
```
Comment (mengqiy) [inline pkg/crd/generator/testData/hack/boilerplate.go.txt]: nit: remove this newline here.

File: pkg/crd/generator/testData/.gitignore (hunk 1 above)
Comment (droot) [reply pkg/crd/generator/testData/.gitignore]: Yes. `testData` is a Kubebuilder generated project.

File: pkg/crd/generator/testData/hack/boilerplate.go.txt (hunk 3 above)
Comment (droot) [reply pkg/crd/generator/testData/hack/boilerplate.go.txt]: Done

File: pkg/crd/generator/testData/Gopkg.toml (hunk 2 above)
Comment (droot) [reply pkg/crd/generator/testData/Gopkg.toml]: removed this file.

File: pkg/crd/generator/testData/.gitignore (hunk 1 above)
Comment (droot) [reply pkg/crd/generator/testData/.gitignore]: removed it.

**Comment by droot** [comment]
@mengqiy addressed the comments. PTAL.

**Pull request merged**